### ISSUE TITLE
Boolean values use lowercase but print in uppercase

### ISSUE THREAD

**Issue by balloonpopper** [godotengine/godot#52060]
### Godot version

3.4 beta 4

### System information

Windows 10

### Issue description

Boolean values must be specified in lowercase 
```
var my_true_var:bool = true
var my_false_var:bool = false
```
Yet when I print these out for debugging purposes they are printed with an uppercase letter
```
print(my_true_var)
True
print(my_false_var)
False
```
For my save/load functions for my game I need to manually correct the case to be able to feed game variables. I believe the case should be preserved so the keywords are correct.

### Steps to reproduce

```
	var mybool:bool=false
	
	print_debug(mybool)
	printerr(mybool)
	printraw(mybool)
	printt(mybool)
	push_error(str(mybool))
	push_warning(str(mybool))
	prints(mybool)
```

**Comment by pycbouh** [comment]
> For my save/load functions for my game I need to manually correct the case to be able to feed game variables. I believe the case should be preserved so the keywords are correct.

Not sure what your serialization process looks like, but you can use `var2str` and `str2var` here instead of manual conversion.

**Comment by balloonpopper** [comment]
Thanks @pycbouh - I didn't know about var2str, that's easier than the conversion I was using.

Correction on the cause - **it appears it's "str" not "print"** changing the case (as seen in debug by "var test = str(mybool)" containing the uppercase version of the boolean). I wouldn't expect "str" to convert the case.

**Comment by pycbouh** [comment]
Yes, it's just a text representation of the boolean type. All objects and data types have their text representation. I'm not sure if there is a good reason to have booleans capitalized like that though.

**Comment by Calinou** [comment]
This is likely a good change to make in 4.0, but I fear there are some people out there relying on the current behavior in `3.x`.

**Comment by balloonpopper** [comment]
I'm not a C programmer unfortunately, but I'm taking an educated guess it's lines 1627-1632 in core/variant/variant.cpp? Looks like "Null" should also be "null"?
```
String Variant::stringify(List<const void *> &stack) const {
	switch (type) {
		case NIL:
			return "null";
		case BOOL:
			return _data._bool ? "true" : "false";
```

**Comment by Calinou** [comment]
> I'm not a C programmer unfortunately, but I'm taking an educated guess it's lines 1627-1632 in core/variant/variant.cpp? Looks like "Null" should also be "null"?
> 
> ```
> String Variant::stringify(List<const void *> &stack) const {
> 	switch (type) {
> 		case NIL:
> 			return "null";
> 		case BOOL:
> 			return _data._bool ? "true" : "false";
> ```
> 
> Instead of
> 
> ```
> String Variant::stringify(List<const void *> &stack) const {
> 	switch (type) {
> 		case NIL:
> 			return "Null";
> 		case BOOL:
> 			return _data._bool ? "True" : "False";
> ```

That seems to be it :slightly_smiling_face: 

And yes, this change should also be made for `null`.

**Comment by balloonpopper** [comment]
I tried to test changing those lines myself but got into compiler hell trying to get MinGW/Scons etc to play nicely on my machine. I'd like to make the change myself so I'm not relying on other people (or committing an untested change to the repository). I'll do a PR if I can work out the compiler issues.
